Apps/WebClient/src/ClientApp/src/components/timeline/TimelineComponent.vue

```diff
@@ -503,19 +509,110 @@ function fetchDataset(entryType: EntryType): Promise<void> {
 }
 
 function fetchTimelineData(): Promise<void | void[]> {
-    const promises = props.entryTypes.map((entryType) =>
-        fetchDataset(entryType)
-    );
+    const cannotAccessEntryTypes: EntryType[] = [];
+
+    const promises: Promise<void>[] = [];
+    props.entryTypes.forEach((entryType) => {
+        {
+            if (canAccessDataset(entryType)) {
+                promises.push(fetchDataset(entryType));
+            } else {
+                cannotAccessEntryTypes.push(entryType);
+            }
+        }
+    });
 
     if (props.commentsAreEnabled) {
         promises.push(retrieveComments(props.hdid));
     }
 
+    showDatasetWarning(cannotAccessEntryTypes);
+
     return Promise.all(promises).catch((err) =>
         logger.error(`Error loading timeline data: ${err}`)
     );
 }
 
+function canAccessDataset(entryType: EntryType): boolean {
+    const dataSource = dataSourceUtil.getDataSource(entryType);
+    const canAccess = !blockedDataSources.value.includes(dataSource);
+    logger.debug(`Can access data source for ${dataSource}: ${canAccess}`);
+    return canAccess;
+}
+
+function showDatasetWarning(entryTypes: EntryType[]): void {
+    const entryTypesLength = entryTypes.length;
+    logger.debug(
+        `Cannot access dataset entry types length: ${entryTypesLength}`
+    );
+
+    if (entryTypesLength > 0) {
+        let warningMessage = "";
+
+        if (entryTypes.length > 1) {
+            warningMessage =
+                "Multiple records are unavailable at this time. Please try again later.";
+        } else {
+            const entryType: EntryType = entryTypes[0];
+            logger.debug(`Setting up warning message for: ${entryType}`);
+            switch (entryType) {
```

Comment:
The entryTypeDetails model has the `name`,

this could be simplified into:

```TypeScript
const name = entryTypeMap.get(EntryType)?.name;
return name ? `${name} are unavailable at this time. Please try again later.` : `Some records may be unavailable at this time, please try again later.`;
```

The fallback message is unlikely to trigger but at least we will have something to report to the users if for any reason the entryTypeMap is incomplete.

Reply:
Changed.